catroid/src/main/java/org/catrobat/catroid/ui/fragment/FragmentFactory.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2017 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.ui.fragment;
+
+import android.app.Fragment;
+import android.content.Context;
+
+final class FragmentFactory {
```

Comment:
very general name if this is only meant to provide BrickCategoryFragment and FormulaEditorCategoryListFragment.
although its named factory, but its not really analog to a GOF factory pattern.

wouldnt it be nicer to use static factory methods in those fragments, that provide the right implementation when called (http://www.androiddesignpatterns.com/2012/05/using-newinstance-to-instantiate.html)